syntax = "proto3";

option go_package = "pb/game";

import "google/protobuf/empty.proto";
import "proto/common/common.proto";
option cc_generic_services = true;

message CentreEnterSceneRequest
{
  SceneInfoPBComponent scene_info = 1;
}

message CentreLeaveSceneRequest
{
}

message CentreLeaveSceneAsyncSavePlayerCompleteRequest
{
}

message CentreSceneInfoRequest {
}


service CentrePlayerScene
{
  rpc EnterScene (CentreEnterSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveScene (CentreLeaveSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveSceneAsyncSavePlayerComplete (CentreLeaveSceneAsyncSavePlayerCompleteRequest) returns (google.protobuf.Empty);
  rpc SceneInfoC2S (CentreSceneInfoRequest) returns (google.protobuf.Empty);
}